### ISSUE TITLE
TypeError: Cannot read property 'getState' of undefined

### ISSUE THREAD

**Issue by Guneetgstar** [joaopaulomoraes/nextjs-with-redux-and-material-ui#9]
I guess the development of this project has been stopped, though I would like to add this a=issue as this project seems very useful to me for someone who wants to bootstrap a next.js project with with redux and material-ui (like me).

Upon running `npm run-script dev` and opening the page at localhost:3000 I got few warning and an error(logs):
```    at new Provider (/**/nextjs-with-redux-and-material-ui/node_modules/react-redux/lib/components/Provider.js:29:25)
    at processChild (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:2995:14)
    at resolve (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:2960:5)
    at ReactDOMServerRenderer.render (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3435:22)
    at ReactDOMServerRenderer.read (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3373:29)
    at renderToString (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3988:27)
    at renderPage (/**/nextjs-with-redux-and-material-ui/node_modules/next/dist/next-server/server/render.js:50:851)
    at Object.ctx.renderPage (webpack-internal:///./pages/_document.js:108:26)
    at Function.getInitialProps (webpack-internal:///./node_modules/next/dist/pages/_document.js:141:19)
    at Function._Document.getInitialProps (webpack-internal:///./pages/_document.js:119:83)
event - build page: /next/dist/pages/_error
wait  - compiling...
event - compiled successfully
/!\ You are using legacy implementaion. Please update your code: use createWrapper() and wrapper.withRedux().
Warning: viewport meta tags should not be used in _document.js's <Head>. https://err.sh/next.js/no-document-viewport-meta
Error: Circular structure in "getInitialProps" result of page "/_error". https://err.sh/vercel/next.js/circular-structure
    at Function.getInlineScriptSource (webpack-internal:///./node_modules/next/dist/pages/_document.js:561:15)
    at NextScript.render (webpack-internal:///./node_modules/next/dist/pages/_document.js:618:28)
    at processChild (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3134:18)
    at resolve (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:2960:5)
    at ReactDOMServerRenderer.render (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3435:22)
    at ReactDOMServerRenderer.read (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3373:29)
    at renderToStaticMarkup (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:4004:27)
    at renderDocument (/**/nextjs-with-redux-and-material-ui/node_modules/next/dist/next-server/server/render.js:3:669)
    at renderToHTML (/**/nextjs-with-redux-and-material-ui/node_modules/next/dist/next-server/server/render.js:52:103)
    at async /**/nextjs-with-redux-and-material-ui/node_modules/next/dist/next-server/server/next-server.js:100:66
Warning: the `Container` in `_app` has been deprecated and should be removed. https://err.sh/vercel/next.js/app-container-deprecated
TypeError: Cannot read property 'getState' of undefined
    at new Provider (/**/nextjs-with-redux-and-material-ui/node_modules/react-redux/lib/components/Provider.js:29:25)
    at processChild (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:2995:14)
    at resolve (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:2960:5)
    at ReactDOMServerRenderer.render (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3435:22)
    at ReactDOMServerRenderer.read (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3373:29)
    at renderToString (/**/nextjs-with-redux-and-material-ui/node_modules/react-dom/cjs/react-dom-server.node.development.js:3988:27)
    at renderPage (/**/nextjs-with-redux-and-material-ui/node_modules/next/dist/next-server/server/render.js:50:851)
    at Object.ctx.renderPage (webpack-internal:///./pages/_document.js:108:26)
    at Function.getInitialProps (webpack-internal:///./node_modules/next/dist/pages/_document.js:141:19)
    at Function._Document.getInitialProps (webpack-internal:///./pages/_document.js:119:83)
     
ready - started server on http://localhost:3000
event - compiled successfully
event - build page: /
wait  - compiling...
event - compiled successfully
/!\ You are using legacy implementaion. Please update your code: use createWrapper() and wrapper.withRedux().
Warning: the `Container` in `_app` has been deprecated and should be removed. https://err.sh/vercel/next.js/app-container-deprecated
TypeError: Cannot read property 'getState' of undefined

**Comment by joaopaulomoraes** [comment]
Hi @Guneetgstar !

Probably this problem is due to the fact that you have updated the version of the `next-redux-wrapper` package, right? I did the clone process and managed to execute the project using the versions prefixed in `yarn.lock`.

As this is a demonstrative project only, it is not really maintained or supported for all new versions. If there is interest, I would be very happy to receive a PR.

**Comment by Guneetgstar** [comment]
Ok I tried to fix it and here is the pull request #10

**Comment by joaopaulomoraes** [comment]
Great job, @Guneetgstar. Merged in #11.
